drivers/gpio/Kconfig.emul

```diff
@@ -0,0 +1,17 @@
+# Emulated GPIO configuration options
+
+# Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+# SPDX-License-Identifier: Apache-2.0
+
+config GPIO_EMUL
+	bool "Emulated GPIO driver"
```

Comment:
Please add ` [EXPERIMENTAL]` to the option title at least for 2.5.  We should get more users and experience with it before deciding whether the driver-specific API is what we're going to want.